webapp/src/js/controllers/reports.js

```diff
@@ -329,24 +331,58 @@ angular
     $scope.$on('VerifyReport', function(e, valid) {
       if ($scope.selected[0].doc.form) {
         $scope.setLoadingSubActionBar(true);
+        const doc = $scope.selected[0].doc;
```

Comment:
So the correct way to do this would be redux?

Reply:
I don't think so. As I understand it redux should be used to store state rather than events. I think the right way would be to replace the `ng-click="emit('VerifyReport', false)"` with something like `ng-click="verifyReport(false)"` which invokes a `$scope.verifyReport` function. Unfortunately because the action bar's scope is InboxCtrl it'd need to be moved there. To keep it clean the actual logic could be pulled out into a service so the code in InboxCtrl could be kept to a minimum.

Eventually I think the action bar will need its own controller...

Reply:
Oh man.. the action-bar has a very hazy separation of concerns and emits 5 other events. Presumably we'll have to get rid of all of these for the angular update. Is there an issue for this already? 

I'm happy to refactor this but would like to do so in its own PR. I'd prefer to make one broader change to the action bar rather than a piecemeal approach for this.

Reply:
There's no issue for this yet. No worries about putting off this refactor.